include/sliding_window/split/reference_metadata.hpp

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#include <seqan3/core/debug_stream.hpp>
+#include <seqan3/io/sequence_file/all.hpp>
+
+#include <iostream>
+#include <fstream>
+
+namespace sliding_window
+{
+
+class reference_metadata
+{
+    public:
+        struct sequence_stats
+        {
+            std::string id;
+            size_t len;
+
+            sequence_stats(std::string ref_id, size_t ref_length)
+            {
+                id = ref_id;
+                len = ref_length;
+            }
+        };
+
+        size_t total_len;
+        std::vector<sequence_stats> sequences;
+
+        reference_metadata(std::filesystem::path filepath, bool from_sequence)
+        {
+            if (from_sequence)
+            {
+                using traits_type = seqan3::sequence_file_input_default_traits_dna;
+                seqan3::sequence_file_input<traits_type> fin{filepath};
+
+                total_len = 0;
+                for (auto & record : fin)
+                {
+                    sequence_stats seq(record.id(), record.sequence().size());
+                    total_len += seq.len;
+                    sequences.push_back(seq);
+                }
```

Comment:
from a style perspective, I would put the two different code branches into own functions and just call them here

```suggestion
            if (from_sequence)
            {
                construct_by_sequence_file(filepath);
            } else
            {
                construct_by_metadata_file(filepath);
            }
```